src/Cassandra.IntegrationTests/Core/PoolShortTests.cs

```diff
@@ -115,5 +117,63 @@ private Task<string[]> ExecuteMultiple(ITestCluster testCluster, Session session
             }
             return tcs.Task;
         }
+        
+        [Test]
+        public void MarkHostDown_PartialPoolConnection()
+        {
+            //start scassandra
+            SCassandraManager.Start();
+            SCassandraManager.SetupInitialConf().Wait();
+            const int connectionLength = 4;
+            var builder = Cluster.Builder()
+                                 .AddContactPoint("127.0.0.1")
+                                 .WithPort(8042)
+                                 .WithPoolingOptions(new PoolingOptions()
+                                     .SetCoreConnectionsPerHost(HostDistance.Local, connectionLength)
+                                     .SetMaxConnectionsPerHost(HostDistance.Local, connectionLength)
+                                     .SetHeartBeatInterval(0));
+            using (var cluster = builder.Build())
+            {
+                var session = (Session)cluster.Connect();
+                var allHosts = cluster.AllHosts();
+
+                TestHelper.WaitUntil(() =>
+                    allHosts.Sum(h => session
+                        .GetOrCreateConnectionPool(h, HostDistance.Local)
+                        .OpenConnections
+                    ) == allHosts.Count * connectionLength);
+                var h1 = allHosts.FirstOrDefault();
+                var pool = session.GetOrCreateConnectionPool(h1, HostDistance.Local);
+                SCassandraManager.DisableConnectionListener().Wait();
+
+                Assert.AreEqual(4, pool.OpenConnections);
+                Assert.IsTrue(h1.IsUp);
+                //remove one connection
+                var ports = SCassandraManager.GetListOfConnectedPorts().Result;
+                SCassandraManager.DropConnection(ports[ports.Length-1]).Wait();
+                Thread.Sleep(500); //wait few miliseconds to update state of driver
+                Assert.AreEqual(3, pool.OpenConnections);
+                Assert.IsTrue(h1.IsUp);
+                //remove one connection
+                ports = SCassandraManager.GetListOfConnectedPorts().Result;
+                SCassandraManager.DropConnection(ports[ports.Length - 1]).Wait();
+                Thread.Sleep(500); //wait few miliseconds to update state of driver
```

Comment:
As in Jenkins is still failing, you can try using `TestHelper.WaitUntil()` to see if its received after a long period of time.
If it isn't, try enabling the heartbeat (to something like 2 seconds).